test/common/http/date_provider_impl_test.cc

```diff
@@ -11,11 +11,63 @@
 #include "gtest/gtest.h"
 
 using testing::NiceMock;
+using testing::Return;
 
 namespace Envoy {
 namespace Http {
 
-TEST(DateProviderImplTest, All) {
+class MockTimeSource : public TimeSource {
+public:
+  MOCK_METHOD(SystemTime, systemTime, ());
+  MOCK_METHOD(MonotonicTime, monotonicTime, ());
+};
+
+TEST(SlowDateProviderImplTest, RequestHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
+  EXPECT_CALL(time_source, systemTime()).WillOnce(Return(time_point));
+
+  SlowDateProviderImpl provider(time_source);
+  TestRequestHeaderMapImpl headers;
+  EXPECT_EQ(nullptr, headers.Date());
+
+  provider.setDateHeader(headers);
+  EXPECT_EQ(headers.Date()->value().getStringView(), "Thu, 01 Jan 1970 00:00:02 GMT");
+}
+
+TEST(SlowDateProviderImplTest, ResponseHeaders) {
+  MockTimeSource time_source;
+  SystemTime time_point{SystemTime::duration(2000000)};
+  EXPECT_CALL(time_source, systemTime()).WillOnce(Return(time_point));
+
+  SlowDateProviderImpl provider(time_source);
+  TestResponseHeaderMapImpl headers;
+  EXPECT_EQ(nullptr, headers.Date());
+
+  provider.setDateHeader(headers);
+  EXPECT_EQ(headers.Date()->value().getStringView(), "Thu, 01 Jan 1970 00:00:02 GMT");
```

Comment:
Interestingly this test seems to only be failing on Windows CI:

```
  headers.Date()->value().getStringView()
    Which is: "Thu, 01 Jan 1970 00:00:00 GMT"
  "Thu, 01 Jan 1970 00:00:02 GMT"
    Which is: 00007FF7C86CE380
Stack trace:
  00007FF7C8621516: (unknown)
  00007FF7C8637464: (unknown)
  00007FF7C8637E93: (unknown)
  00007FF7C86216CC: (unknown)
... Google Test internal frames ...
```
Any ideas on why that might be the case? It passes locally for me on macOS, and unfortunately I don't have a Windows machine to debug with

Reply:
@wrowe @sunjayBhatia another Windows CI failure question. Thanks :)

Reply:
taking a look now

Reply:
thank you!

Reply:
having some local build issues, still working it out